### ISSUE TITLE
Update vulnerable dependencies

### ISSUE THREAD

**Issue by admitt** [microsoft/spring-cloud-azure#669]
## Expected Behavior
While running `mvn org.owasp:dependency-check-maven:check` I expect no vulnerable libs to be present in the dependencies of my project

## Current Behavior
Those dependencies have known vulnerabilities

spring-cloud-starter-azure-eventhubs-1.2.3.jar: CVE-2018-1272, CVE-2016-9878, CVE-2018-1271, CVE-2018-1270
spring-integration-eventhubs-1.2.3.jar: CVE-2019-3772, CVE-2018-1272, CVE-2016-9878, CVE-2018-1271, CVE-2018-1270
spring-integration-azure-core-1.2.3.jar: CVE-2019-3772
netty-transport-4.1.45.Final.jar: CVE-2020-11612
adapter-rxjava-2.4.0.jar: CVE-2018-1000844, CVE-2018-1000850

## Possible Solution
Update libraries if possible without breaking the current logic

## Steps to Reproduce (for bugs)
* include spring-cloud-azure-eventhubs-stream-binder in your pom.xml
* run `mvn org.owasp:dependency-check-maven:check`
* check results in console and a report in target/dependency-check-report.html

## Snapshot Code for Reproduce 
```xml
        <dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>spring-cloud-azure-eventhubs-stream-binder</artifactId>
            <version>1.2.3</version>
        </dependency>
```

## Branch
* master

## Your Environment
* Version used: 1.2.3
* Operating System and version (desktop or mobile): Mac OS
* SDK version: spring-cloud-azure-eventhubs-stream-binder

**Comment by saragluna** [comment]
Thanks for reporting this and we'll look into it.

**Comment by yiliuTo** [comment]
There are some vulnerable dependencies in Azure SDK repo, and we have reported this issue. Thank you.

**Comment by admitt** [comment]
Great! Thanks for the update!

**Comment by yiliuTo** [comment]
We have updated the Spring Boot version and Azure version for some of vulnerable dependencies in version 1.2.4. For the rest imported in Azure SDK, we have reported this issue https://github.com/Azure/azure-sdk-for-java/issues/10595.

**Comment by yiliuTo** [comment]
For the vulnerable dependencies imported in Azure SDK, they will be resolved when https://github.com/Azure/azure-sdk-for-java/pull/10531 is merged in as it will be upgrading the Netty versions.

**Comment by hemalathamoorthy** [comment]
Is there any update on this issue?

**Comment by yiliuTo** [comment]
@hemalathamoorthy The vulnerable dependencies have been updated in the Azure SDK repo, and we will update them in the next release, perhaps 3 weeks later.

**Comment by admitt** [comment]
Hi, any updates? Hope the upgrade made it to the release :) Thanks!

**Comment by yiliuTo** [comment]
Hi @admitt , the upgrade of vulnerable dependencies have been released in version 1.2.6 except the spring-framework and spring-integration. As for the mentioned 2 dependencies, the versions we used are not listed in the versions that have those vulnerabilities thus we didn't upgrade them, perhaps there are some bugs with the vulnerability checking tool.

**Comment by saragluna** [comment]
Closing this now.
